api/v3/Membership.php

```diff
@@ -33,6 +33,23 @@
  */
 
 /**
+ * Adjust Metadata for Delete action.
+ *
+ * The metadata is used for setting defaults, documentation & validation.
+ *
+ * @param array $params
+ *   Array of parameters determined by getfields.
+ */
+function _civicrm_api3_membership_delete_spec(&$params) {
+  $params['preserve_contribution'] = array(
+    'api.required' => 0,
+    'title' => 'Preserve Contribution',
+    'description' => 'By default this is 0, or 0 if not set. Set to 1 to preserve the associated contribution record when membership is deleted.',
```

Comment:
Actually I'm really really happy to change the default to preserve by default. In fact at the API level I'm OK with us not supporting delete the contribution as well at all.

We'd have to announce it but I feel the existing behaviour is dangerous